Cargo.toml

```diff
@@ -42,7 +42,15 @@ features = ["raw_value"]
 ### FEATURES #################################################################
 
 [features]
-default = []
+default = ["std"]
+
+std = ["serde/std", "itoa/std"]
+
+# Provide integration for heap-allocated collections without depending on the
+# rest of the Rust standard library.
+# NOTE: Disabling both `std` *and* `alloc` features is not supported yet.
```

Comment:
Please find a way to produce a reasonable error message for this case. The current error if you do `serde_json = { default-features = false }` is \>1500 lines which is insane.

https://github.com/dtolnay/syn/blob/1.0.14/tests/features/mod.rs + https://github.com/dtolnay/syn/blob/1.0.14/tests/features/error.rs shows one good way.

Reply:
Oh, that's neat! I wanted to do so but couldn't think of a reliable way to only show one compilation error.

As written [here](https://github.com/serde-rs/json/pull/606/files#r369071931), maybe it'd be good to just support `std` and no-`std` (implying using `alloc`)? This would mean we wouldn't have to care about detecting no `std` and `alloc` anymore.

Reply:
Done in https://github.com/serde-rs/json/pull/606/commits/40b984cfcca926f2c0cc176b3bcf91b06fa6460f.

Is there a way to run trybuild with custom feature flags, like compiletest? It'd be quite handy to have a UI test for `--no-default-features` so that we can catch whenever the error message regresses.

Reply:
> The current error if you do `serde_json = { default-features = false }` [...]

Shouldn't this new error be considered a breaking change?